src/directives/click-outside.js

```diff
@@ -1,31 +1,44 @@
 function directive (e, el, binding, v) {
+  if (!v.context.isActive) return
   let cb = () => true
 
   if (binding.value) cb = binding.value
 
-  if (v.context.isActive &&
-    (e && e.target) &&
-    (e.target !== el && !el.contains(e.target)) &&
-    cb(e)
-  ) {
-    v.context.isActive = false
+  const elements = (binding.include || (() => []))()
+  elements.push(el)
+
+  if (!clickedInEls(e, elements) && cb(e)) {
+    setTimeout(() => v.context.isActive = false, 0)
+  }
+}
+
+function clickedInEls (e, elements) {
+  for (const el of elements) {
+    if (clickedInEl(e, el)) return true
   }
+  return false
+}
+
+function clickedInEl (e, el) {
+  e = e || {}
+  const b = el.getBoundingClientRect()
+  const x = e.clientX
+  const y = e.clientY
+  return x >= b.left && x <= b.right && y >= b.top && y <= b.bottom
 }
 
 export default {
   name: 'click-outside',
 
   bind (el, binding, v) {
     v.context.$vuetify.load(() => {
-      const outside = document.querySelector('[data-app]')
       const click = e => directive(e, el, binding, v)
```

Comment:
I understand it was like this before, but since you are in here and I'm trying to move things over to a common syntax, all event methods and variables that reference events should be pre-pended with **on**. So in this event, it would be **onClick** as the variable name.